screens/CheckoutScreen.js

```diff
@@ -0,0 +1,321 @@
+import React from 'react';
+import {
+  Alert,
+  AsyncStorage,
+  Text,
+  TouchableOpacity,
+  View
+} from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+import ProductCartCard from '../components/ProductCartCard';
+import ProductDisplayCard from '../components/ProductDisplayCard';
+import {
+  addTransaction,
+  categories,
+  getUser,
+  loadProductsData,
+  updateCustomerPoints
+} from '../lib/checkoutUtils';
+import { FlatListContainer } from '../styled/checkout.js';
+import { ScrollCategory, TextHeader } from '../styled/shared';
+
+// TODO figure out how to break these state-modifying functions into helper files?
+// TODO note edge cases that fail with current workflow
+
+// TODO research styling for tablets to be both horizontal and vertical friendly
+export default class ClerkProductsScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      customer: null,
+      fullProducts: [],
+      products: [],
+      filter: null,
+      cart: [],
+      isLoading: true,
+      totalPrice: 0,
+      totalPoints: 0,
+      rewardsApplied: 0,
+      rewardsAvailable: 0
+    };
+  }
+
+  async componentDidMount() {
+    const customerId = await AsyncStorage.getItem('customerId');
+    const customer = await getUser(customerId);
+    const productsData = await loadProductsData();
+
+    this.setState({
+      customer: customer,
+      rewardsAvailable: customer.rewardsAvailable,
+      fullProducts: productsData,
+      products: productsData,
+      isLoading: false
+    });
+  }
+
+  // Adds one item of product type to cart.
+  addToCart(item) {
+    item.cartCount++;
+    var currentCart = this.state.cart;
+    var currentItem = currentCart.filter(product => product.id == item.id);
+    var totalPrice = this.state.totalPrice;
+    totalPrice = totalPrice + item.customerCost;
+    if (currentItem.length == 0) {
+      currentCart.push(item);
+    }
+    this.setState({
+      cart: currentCart,
+      totalPrice: totalPrice
+    });
+  }
+
+  // Subtracts one item of product type from cart.
+  removeFromCart(item) {
+    item.cartCount--;
+    var currentCart = this.state.cart;
+    currentCart = currentCart.filter(item => item.cartCount > 0);
+    var totalPrice = this.state.totalPrice;
+    totalPrice = totalPrice - item.customerCost;
+    this.setState({
+      cart: currentCart,
+      totalPrice: totalPrice
+    });
+  }
+
+  handleCategoryPress = filter => {
+    const toSet =
+      filter == 'All'
+        ? this.state.fullProducts
+        : this.state.fullProducts.filter(product =>
+            product.category.includes(filter)
+          );
+    this.setState({ products: toSet });
+  };
+
+  // TODO why is this async
```

Comment:
Wait yeah I forget why I made this a Promise...since I work with just state I think it's ok to remove the Promise...